packages/block-editor/src/store/reducer.js

```diff
@@ -1645,6 +1645,33 @@ export function highlightedBlock( state, action ) {
 	return state;
 }
 
+/**
+ * Reducer returning hovered blocks state.
+ *
+ * @param {Object} state  Current hovered blocks state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export function hoveredBlocks(
+	state = { hoveredBlockIds: [], eventData: {} },
+	action
+) {
+	switch ( action.type ) {
+		case 'SET_HOVERED_BLOCKS':
+			const { hoveredBlockIds, timeStamp, mouseCoords } = action;
+			return {
+				hoveredBlockIds,
+				eventData: {
```

Comment:
Why do we need thisi information in state?

Reply:
the scenario is: "move mouse over a block which is deeply nested". This will trigger N hover events for N nested blocks. If we have a ton of those blocks, we'll be dispatching to the store N times, which caused performance issues. To avoid updating the store that often, we need to compare the timestamp of an event with the timestamp of another event. So we track the timestamp here. I think the other option would be storing it in a global variable in the file where it's used

Reply:
Is there a way to "stop propagation" somehow on the event?

Reply:
i'm unsure, I'll leave that for @Addison-Stavlo to answer

Reply:
> Is there a way to "stop propagation" somehow on the event?

I didn't have any luck.  It's not one event bubbling, but many events being triggered simultaneously.

I think we can get around requiring this information and extra complexity by debouncing. e41e82d

Reply:
That said, while having a way to track which blocks are hovered is neat, given the comments from design below it doesn't seem like we have a consensus on where we should use this information yet (probably not in the document actions header from the sounds of it 😆 ).  This could prove useful in implementing something similar in the future (such as closer to the cursor https://github.com/WordPress/gutenberg/pull/25348#issuecomment-710303993), but until we figure out the right place to utilize this there is no rush for adding the handlers and store either.